eth/tracers/native/prestate.go

```diff
@@ -135,12 +166,64 @@ func (t *prestateTracer) CaptureTxStart(gasLimit uint64) {
 	t.gasLimit = gasLimit
 }
 
-func (t *prestateTracer) CaptureTxEnd(restGas uint64) {}
+func (t *prestateTracer) CaptureTxEnd(restGas uint64) {
+	if !t.config.CollectPost {
+		return
+	}
+
+	for addr, state := range t.pre {
+		// the deleted account's state is pruned
+		if _, ok := t.deleted[addr]; ok {
+			continue
+		}
+		modified := false
+		postAccount := &account{Storage: make(map[common.Hash]common.Hash)}
+		newBalance := bigToHex(t.env.StateDB.GetBalance(addr))
+		newNonce := t.env.StateDB.GetNonce(addr)
+		newCode := bytesToHex(t.env.StateDB.GetCode(addr))
+
+		if newBalance != t.pre[addr].Balance {
+			modified = true
+			postAccount.Balance = newBalance
+		}
+		if newNonce != t.pre[addr].Nonce {
+			modified = true
+			postAccount.Nonce = newNonce
+		}
+		if newCode != t.pre[addr].Code {
+			modified = true
+			postAccount.Code = newCode
+		}
+
+		for key, val := range state.Storage {
```

Comment:
This will capture storage modifications. Can you please extend it for new storage slots and slot clearing?

- Re clearing: I think `GetState` will return `common.Hash{}` if a slot is empty, then we shouldnt include it in `post`
- Re new slot: We can watch `SSTORE` opcode and keep track of new slots for every contract. Then we can add them also to `post`. Note that if a revert happens after those slots are created they will be wiped, but if you do a `GetState` here and check it against the empty value that should guard against this.

Reply:
- Re clearing: thanks for the advice, I'll clear if empty;
- Re new slot: those slots were included in the `pre`, and thanks for the `revert`  behavior